index.d.ts

```diff
@@ -6,12 +6,23 @@ import {
   SectionListData,
   StyleProp,
   RefreshControlProps,
-} from 'react-native';
+  SectionListProps,
+  FlatListProps
+} from 'react-native'
+
+// Copy from TS 3.5
+type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>
+
+// Original flat list component props
+type FlatListAllowedProps<ItemType = any> = Omit<FlatListProps<ItemType>,
+  | "data"
+  | "renderItem"
+>
 
 /**
  * React Native Super Grid Properties
  */
-export interface FlatGridProps<ItemType = any> {
+export interface FlatGridProps<ItemType = any> extends FlatListAllowedProps<ItemType> {
```

Comment:
If we can extend from FlatListProps then there is no need to explicitly mention props like `ListEmptyComponent`, `onEndReached`...etc here

Reply:
I've been reviewing the types now, and there are some diffs that seems wrong to me like:
`keyExtractor?: (item: ItemType[], index: number) => string;` - declared on the super grid
`keyExtractor?: (item: ItemT, index: number) => string;` - declared on the rn grid
I'll keep the RN as a source of truth and remove the types that have the same signature or seems to have a wrong one from the lib